### DISCUSSION TITLE
Adding new CSS variables to component

### DISCUSSION THREAD

**Discussion by vishnup-ditto** [mantinedev/mantine#5821] [Q&A]
Starting a discussion since I couldn't find a neat way to do with this mantine. I will try to illustrate the requirement using a simple hypothetical example. 
Suppose I have a `<BadgeWithLeftSection/>` component that I have abstracted and created mantine component `<Badge />`

```
// JSX
const BadgeWithLeftSection = ({iconColor: string}) => {
	return (
		<Badge
		leftSection={<Icon />}	
		classNames={{
			section: badgeClasses.section
		}}
		// all other props
		/>
}

// CSS

.section {
	&[data-position="left"] {
		svg {
			color: [--badge-icon-color] // **this is the requirement, set this color based on the prop I receive in the component!**
		}	
	}
}

```

Basically, if I receive a color(or similar) prop at the component, I am interested in a way to pass this to CSS as a CSS custom property.  

NB: I know this example is a bad representation of the problem (lot's of ways in which this could be refactored better). I am hoping this just serves as a quick illustration. The question is, is there a mantine way of doing this as I currently work around using `element.style.setProperty` to do this now and it doesn't feel too clean.

**Comment by tombohub** [comment]
For start I don't think thats valid css: `color: [--badge-icon-color]` should be `color: var(--badge-icon-color)`

If you really want to use css variable then you can set the value of css variable as documented in: https://mantine.dev/styles/styles-overview/#style-prop

example:
```javascript
import { Box } from '@mantine/core';

function Demo({ color }: { color: string }) {
  // Later you will be able to use var(--my-color) in any nested element
  return <Box style={{ '--my-color': color }}>My box</Box>;
}
```

See if that works

**Reply by vishnup-ditto** [reply]
Thanks for the reply. Aah, I meant the `icon-color` only as an example. 

This fits my needs. This is okay. I ended up finding another `_vars` props at mantine component level that has a typedef comment that it is used to add css variables at root. Any idea about that?

**Reply by tombohub** [reply]
don't know much about `__vars`. It's for internal use only seems like it
